apps/main/lib/main/repositories/articles_repo.rb

```diff
@@ -0,0 +1,14 @@
+require "blog/repository"
+require "main/entities/article"
+
+module Main
+  module Repositories
+    class ArticlesRepo < Blog::Repository[:articles]
+      struct_namespace Main::Entities
```

Comment:
`struct_namespace` belong to rom-rb https://github.com/rom-rb/rom/blob/667dee8683e39c8f18f92e3ae295f3a4ef3f0164/repository/lib/rom/repository.rb#L75 rom returns data from the DB wrapped in a struct class.

By default if you do not specify anything it will return a struct which `class` would be `ROM::Struct::#{your relation in singular}` - for example `ROM::Struct::Article`.

What `struct_namespace` allow you is a way to specify to rom where to look for custom struct classes that will wrap the returning data - in this project they live under `apps/main/lib/main/entities/` folder https://github.com/dry-rb/dry-web-blog/blob/7f3d10ae9c963f7921471a6561ee8b05895e85b7/apps/main/lib/main/entities/article.rb